src/cgw_connection_processor.rs

```diff
@@ -135,6 +146,20 @@ impl CGWConnectionProcessor {
         self.serial = Some(evt.serial.clone());
         let device_type = CGWDeviceType::from_str(caps.platform.as_str()).unwrap();
 
+        // Check if device queue already exist
+        // If yes - it could mean that we have device reconnection event
+        // The possible reconnect reason could be: FW Upgrade or Factory reset
+        // Need to make sure queue is unlocked to process requests
+        // If no - create new message queue for device
+        let queue_lock = CGW_MESSAGES_QUEUE.read().await;
+        if queue_lock.check_messages_queue_exists(&evt.serial).await {
```

Comment:
Replace with create_device_messages_queue completely;
This function is safe (after rework) to invoke it even if queue exists;

Reply:
I've missed the explanation, sorry; Ignore previous comment